File: challenge/src/main/java/com/meli/challenge/domain/Product.java
package com.meli.challenge.domain;


import lombok.*;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
public class Product {
    private String id;
    private String title;
    private String description;
    private double price;
    private String currency;
    private List<String> images;
    private SellerInfo sellerInfo;
    private List<String> paymentMethods;
    private int stock;
    private AdditionalDetails additionalDetails;
}

File: meli/src/app/domain/byId/Product.ts
import { AdditionalDetails } from './AdditionalDetails';
import { SellerInfo } from './SellerInfo';

export interface Product {
  id: string;
  title: string;
  description: string;
  price: number;
  currency: string;
  images: string[];
  sellerInfo: SellerInfo;
  paymentMethods: string[];
  stock: number;
  additionalDetails: AdditionalDetails;
}
